exercise_2/WeightMatrix.cpp

```diff
@@ -1,24 +1,33 @@
 #include "WeightMatrix.h"
 #include <iostream>
 
-WeightMatrix::WeightMatrix() {}
-
-WeightMatrix::WeightMatrix(const int& vertices)
+WeightMatrix::WeightMatrix(const int& vertices_)
+:vertices{vertices_}
 {
-	std::vector<std::vector<int>> matrix(vertices, std::vector<int>(vertices));
-	this->weightMatrix = matrix;
+	std::vector<std::vector<int>> matrix_(vertices_, std::vector<int>(vertices));
+	weightMatrix = matrix_;
 }
 
 WeightMatrix::~WeightMatrix() = default;
 
-std::ostream& operator << (std::ostream& out, const WeightMatrix& m)
+int WeightMatrix::GetMatrixElement(int& start, int& end)
+{
+	return weightMatrix[start][end];
+}
+
+void WeightMatrix::SetMatrixElement(int& start, int& end, int& weight) 
```

Comment:
pass primitives by copy unless you it's an output parameter and you want to change its value in the function. Which is hardly ever used.
Remember to use const& when passing objects.